pkg/api/playlist_play.go

```diff
@@ -68,23 +70,33 @@ func (hs *HTTPServer) populateDashboardsByTag(ctx context.Context, orgID int64,
 
 // Deprecated -- the frontend can do this better
 func (hs *HTTPServer) LoadPlaylistDashboards(ctx context.Context, orgID int64, signedInUser *user.SignedInUser, playlistUID string) (dtos.PlaylistDashboardsSlice, error) {
-	playlistItems, _ := hs.LoadPlaylistItems(ctx, playlistUID, orgID)
+	dto, err := hs.playlistService.GetWithItems(ctx,
+		&playlist.GetPlaylistByUidQuery{UID: playlistUID, OrgId: orgID})
+	if err != nil {
+		return nil, err
+	}
+
+	playlistItems := dto.Items
 
 	dashboardByIDs := make([]int64, 0)
 	dashboardByTag := make([]string, 0)
 	dashboardIDOrder := make(map[int64]int)
 	dashboardTagOrder := make(map[string]int)
 
-	for _, i := range playlistItems {
-		if i.Type == "dashboard_by_id" {
-			dashboardID, _ := strconv.ParseInt(i.Value, 10, 64)
+	for i, item := range playlistItems {
+		if item.Type == "dashboard_by_id" {
+			dashboardID, _ := strconv.ParseInt(item.Value, 10, 64)
 			dashboardByIDs = append(dashboardByIDs, dashboardID)
-			dashboardIDOrder[dashboardID] = i.Order
+			dashboardIDOrder[dashboardID] = i
+		}
+
+		if item.Type == "dashboard_by_tag" {
+			dashboardByTag = append(dashboardByTag, item.Value)
+			dashboardTagOrder[item.Value] = i
 		}
 
-		if i.Type == "dashboard_by_tag" {
-			dashboardByTag = append(dashboardByTag, i.Value)
-			dashboardTagOrder[i.Value] = i.Order
+		if item.Type == "dashboard_by_uid" {
+			return nil, fmt.Errorf("dashboard_by_uid not supported by this deprecated API")
```

Comment:
obviously we could implement it... but it seems better to throw an error.  If we do not hear any complaints about this we can then actually remove query endpoint